### PR TITLE
fix: prevent array splice notification on non-mutating splice() calls

### PR DESCRIPTION
**Pull request by nicholasrice**
# Pull Request

## 📖 Description
This PR fixes a bug where array observer would notify a splice when the mutation that occurred did not alter the array values. It appears in FE2 migration, an attempt to optimize for single-splice processing short-cut the projection process, which would detect non-mutation scenarios like splicing an array with the same values. 

This issue has significant impact of the `repeat` directive, causing it to disconnect and re-connect views un-necessarily. 

### 🎫 Issues
fixes ##6431

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Review by EisenbergEffect — APPROVED** [review]
Doh! This is my fault. I thought I had a fast path opportunity there and didn't realize there was this particular side-effect.

File: packages/web-components/fast-element/src/observation/arrays.spec.ts [hunk 1]
```diff
@@ -433,6 +433,26 @@ describe("The ArrayObserver", () => {
...
+
+    it("should not deliver splices for .splice() when .splice() does not change the items in the array", async () => {
+        ArrayObserver.enable();
+        const array = [1,2,3,4,5];
+        const observer = Observable.getNotifier(array);
+        let splices;
+
+        observer.subscribe({
+            handleChange(source, args) {
+                console.log(args[0])
```
Comment (EisenbergEffect) [inline packages/web-components/fast-element/src/observation/arrays.spec.ts:445]: Did you mean to leave the `console.log()` here?

Comment (nicholasrice) [reply packages/web-components/fast-element/src/observation/arrays.spec.ts:445]: sure didn't!

Comment (nicholasrice) [inline packages/web-components/fast-element/src/observation/arrays.spec.ts:445]:
```suggestion
```

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**